lib/discourse_ip_info.rb

```diff
@@ -0,0 +1,39 @@
+require_dependency 'maxminddb'
+
+class DiscourseIpInfo
+  include Singleton
+
+  def initialize
+    begin
+      @mmdb_filename = File.join(Rails.root, 'vendor', 'data', 'GeoLite2-City.mmdb')
+      @mmdb = MaxMindDB.new(@mmdb_filename, MaxMindDB::LOW_MEMORY_FILE_READER)
```

Comment:
Like @SamSaffron suggested, have you run `memory_profiler` on this? 
How many objects does this add in low memory vs default mode?

Reply:
Default reader:
```
# Total allocated: 5178841 bytes (118759 objects)
# Total retained:  121235 bytes (2115 objects)
```

Low memory reader:
```
# Total allocated: 62659823 bytes (118756 objects)
# Total retained:  57733098 bytes (2112 objects)
```

There is a very small difference in the number of objects because there is a big one that holds the whole MaxMind database in memory.

Reply:
Yeah, we definitely want the low memory version ;) We can take the 3ms since it's such a low traffic endpoint anyways

Reply:
It depends. I believe, on average, this is going to be called for three times for every user that accesses his preferences page (which in my opinion is something more than a low traffic endpoint). Anyway, some of the calls will be cached anyway and after all, 3ms it's not that bad.